Issue2023curlybracket

AxelTWC opened the pull request:
Pull Request for #2023 
## Issue2023curlybracket

This code currently raises a warning and detects the following Extra Brackets:

- OPEN_ROUND_BRACKETS
- CLOSE_ROUND_BRACKETS
- OPEN_CURLY_BRACKETS
- CLOSE_CURLY_BRACKETS 

and outputs the warning number 1016. 

In order to remove the warning, the user would need to look at the warning message showing the line number of the extra bracket.

## Note
- For Developers, Quick Build the Compiler twice as there is an edit in the Umple Internal Parser Code.
- Multiple Test Cases has been added to test and solve this issue. 
- An extra user manual guideline has been added to show how to reproduce this warning and how to solve the warning.

TimLethbridge committed:
Fixing spelling of Overvlow

TimLethbridge committed:
Added new error 1020 for extraneous curly bracket

AxelTWC committed:
Round Bracket Detection and Curly Bracket Not Detecting

Currently the code detects the Open and Close Round Brackets and have implementations to create the variable OPEN_CURLY_BRACKET and CLOSE_CURLY_BRACKET, however the grammar does not recognize the curly brackets yet.

AxelTWC committed:
Included Test Cases And User Documentation

Included Test Cases And User Documentation for Issue 2023 Curly Bracket Detection

AxelTWC committed:
A small error was made previously.

A Small Error on the Test Cases was made previously where all "class" was "Class"

AxelTWC committed:
Remove Test Case 5

Remove Test Case 5

AxelTWC committed:
Add Ignore Test

Ignoring Test since the Parser needs to compile twice and Jenkins only compile the Compiler once.

AxelTWC committed:
Moved Ignore Annotation

Moved Ignore Annotation

TimLethbridge committed:
Merge branch 'master' into issue2023curlybracket

TimLethbridge commented on cruise.umple/src/class/umple_classes.grammar:
```diff
@@ -29,12 +29,16 @@ associationClassDefinition : associationClass [name] { [[associationClassContent
 innerStaticClass : static [[classDefinition]]
 innerNonStaticClass : inner [[classDefinition]]
 
+openCurlBracket- : {
```
Please remove these two unneeded rules

TimLethbridge commented on cruise.umple/src/class/UmpleInternalParser_CodeClass.ump:
```diff
@@ -64,7 +64,9 @@ class UmpleInternalParser
     }
-
+    //Reveal the current class token
+    //System.out.print("Token:" + t);
```
Please take out this debug code

TimLethbridge commented on cruise.umple/src/class/UmpleInternalParser_CodeClass.ump:
```diff
@@ -352,7 +357,8 @@ class UmpleInternalParser
       return;
     }
-
+    //Reveals the current detected class token
```
Please take out this debug code

TimLethbridge commented on cruise.umple/src/class/UmpleInternalParser_CodeClass.ump:
```diff
@@ -523,6 +529,9 @@ class UmpleInternalParser
     {
       analyzeException(token,aClass);
     }
+    else if (token.is("checkForUnintendedBracket")){
```
This check should only be at the top level ... and hence should only be there once ...

TimLethbridge commented on cruise.umple/src/class/umple_classes.grammar:
```diff
@@ -16,7 +16,7 @@ externalDefinition : external [=interface]? [name] { [[classContent]]* }
 interfaceDefinition : interface [name] { [[depend]]* [[interfaceBody]] }
 
-// Associations can be declared outside the body of classes. 
+// Associations can be declared outside the body of classes.
```
Please reset so there is no change to the space here

TimLethbridge commented on cruise.umple/src/class/umple_classes.grammar:
```diff
@@ -29,12 +29,16 @@ associationClassDefinition : associationClass [name] { [[associationClassContent
 
 // The following items can be found inside the body of classes or association classes
-classContent- : [[comment]] | [[reqImplementation]] | [[innerClass]] | [[mixsetDefinition]] | [[distributable]] | [[proxyPattern]] | [[strictness]] | [[classDefinition]] | [[trace]] | [[emitMethod]] | [[templateAttributeDefinition]] | [[primitiveDefinition]] | [[portDefinition]] | [[portBindingDefinition]] | [[position]] | [[displayColor]] | [[abstract]] | [[keyDefinition]] | [[softwarePattern]]  ...
+classContent- :  [[comment]] | [[reqImplementation]] | [[innerClass]] | [[mixsetDefinition]] | [[distributable]] | [[proxyPattern]] | [[strictness]] | [[classDefinition]] | [[trace]] | [[emitMethod]] | [[templateAttributeDefinition]] | [[primitiveDefinition]] | [[portDefinition]] | [[portBindingDefinition]] | [[position]] | [[displayColor]] | [[abstract]] | [[keyDefinition]] | [[softwarePattern]] ...
```
Please reset so there is no change to the line ... it is just a space

TimLethbridge commented on cruise.umple/src/en.error:
```diff
@@ -207,7 +207,8 @@
 1015: 3, "http://manual.umple.org/?BasicStateMachines.html", Methods in standalone state machines might not be correctly generated. State machines can be used in traits or put directly in a class to achieve the same effect. ;
-
+1016: 3, "http://manual.umple.org/?PageBeingDeveloped.html", Extra Open Or Closing Bracket encountered. Processing terminated. All code beyond this point is ignored. ;
+1020: 3, "http://manual.umple.org/?PageBeingDeveloped.html", Extra closing curly bracket encountered. Processing terminated. All code beyond this point is ignored. ;
```
Please take out 1020 and change 1016 so the 'Processing terminated and following is no longer there.

AxelTWC committed:
Code Review Changes

This push includes the necessary review changes from previous requested changes.
It also changed the user manual to not have a slash as it was giving errors.

AxelTWC committed:
Small Changes to the html naming file

Small Changes to the html naming file
Located in en.error.

AxelTWC committed:
Update HTML Name back to original

Update HTML Name back to original to pass AppVeyor requirements.
